### PR TITLE
[RLlib] break up the learner group tests into shorter tests

### PR DESCRIPTION
**Pull request by avnishn**
break them up into 3 classes  because they all take a very long time to run
causing timeouts on master

Signed-off-by: Avnish <avnishnarayan@gmail.com>

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/learner/tests/test_learner_group.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def local_training_helper(self, fw, scaling_mode) -> None:
         check(local_learner.get_state(), learner_group.get_state())
 
 
-class TestLearnerGroup(unittest.TestCase):
+class TestLearnerGroup1(unittest.TestCase):
```
Comment (kouroshHakha) [inline rllib/core/learner/tests/test_learner_group.py:112]: rename: TestLearnerGroupSyncUpdate

File: rllib/core/learner/tests/test_learner_group.py [hunk 2]
```diff
@@ -497,6 +450,69 @@ def test_load_module_state_errors(self):
             del learner_group
 
 
+class TestLearnerGroup3(unittest.TestCase):
```
Comment (kouroshHakha) [inline rllib/core/learner/tests/test_learner_group.py:453]: rename: TestLearnerGroupAsyncUpdate

File: rllib/core/learner/tests/test_learner_group.py [hunk 3]
```diff
@@ -246,60 +246,13 @@ def test_add_remove_module(self):
...
-                loss = np.mean(
-                    [res[ALL_MODULES][Learner.TOTAL_LOSS_KEY] for res in results]
-                )
-                min_loss = min(loss, min_loss)
-                print(f"[iter = {iter_i}] Loss: {loss:.3f}, Min Loss: {min_loss:.3f}")
-                # The loss is initially around 0.69 (ln2). When it gets to around
-                # 0.57 the return of the policy gets to around 100.
-                if min_loss < 0.57:
-                    break
+class TestLearnerGroup2(unittest.TestCase):
```
Comment (kouroshHakha) [inline rllib/core/learner/tests/test_learner_group.py:250]: rename TestLearnerGroupCheckpointRestore

**Comment by avnishn** [comment]
addressed comments

File: rllib/core/learner/tests/test_learner_group.py (hunk 3 above)
Comment (avnishn) [reply rllib/core/learner/tests/test_learner_group.py]: done

File: rllib/core/learner/tests/test_learner_group.py (hunk 2 above)
Comment (avnishn) [reply rllib/core/learner/tests/test_learner_group.py]: done

File: rllib/core/learner/tests/test_learner_group.py (hunk 1 above)
Comment (avnishn) [reply rllib/core/learner/tests/test_learner_group.py]: done

File: rllib/core/learner/tests/test_learner_group.py [hunk 4]
```diff
@@ -128,7 +128,7 @@ def test_learner_group_local(self):
 
     def test_update_multigpu(self):
         fws = ["torch", "tf2"]
-        scaling_modes = REMOTE_SCALING_CONFIGS.keys()
+        scaling_modes = ["multi-gpu-ddp", "remote-gpu"]
```
Comment (kouroshHakha) [inline rllib/core/learner/tests/test_learner_group.py:131]: @avnishn Can you add explanations on why these changes are made? `REMOTE_SCALING_CONFIGS.keys()` --> `["multi-gpu-ddp", "remote-gpu"]`?

File: .buildkite/pipeline.gpu_large.yml [hunk 5]
```diff
@@ -27,7 +27,7 @@
     # --jobs 2 is necessary as we only need to have at least 2 gpus on the machine
```
Comment (kouroshHakha) [inline .buildkite/pipeline.gpu_large.yml:27]: comment still says --jobs 2 is necessary

Comment (avnishn) [reply .buildkite/pipeline.gpu_large.yml:27]: yep I was exploring here. by removing jobs=2

I'm trying to understand where my cuda oom comes from

Comment (avnishn) [reply .buildkite/pipeline.gpu_large.yml]: added this backl

File: rllib/core/learner/tests/test_learner_group.py (hunk 4 above)
Comment (avnishn) [reply rllib/core/learner/tests/test_learner_group.py:131]: We can reduce the number of tests that we need to run while getting full test coverage by doing this.

**Review by kouroshHakha — APPROVED** [review]
